File: src/employeemanager/tasks/FileTask.java
package employeemanager.tasks;

import employeemanager.model.Employee;
import employeemanager.repository.EmployeeRepository;
import employeemanager.exceptions.AppException;

import java.util.List;

public class FileTask implements Runnable {

    public enum Mode { READ, WRITE }

    private String filePath;
    private Mode mode;
    private List<Employee> employees;
    private EmployeeRepository repository;

    public FileTask(String filePath,
                    Mode mode,
                    List<Employee> employees,
                    EmployeeRepository repository) {
        this.filePath = filePath;
        this.mode = mode;
        this.employees = employees;
        this.repository = repository;
    }

    @Override
    public void run() {
        try {
            switch (mode) {
                case READ -> repository.loadEmployees(filePath);
                case WRITE -> repository.saveEmployees(filePath, employees);
            }
        } catch (AppException ex) {
            System.err.println("FileTask error (" + mode + "): " + ex.getMessage());
        }
    }
}

File: Part 3/EmployeeManager/tasks/FileTask.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using EmployeeManager.Models;
using EmployeeManager.Repository;
using EmployeeManager.Exceptions;

namespace EmployeeManager.Tasks
{
    public class FileTask
    {
        public enum Mode { READ, WRITE }

        private readonly string filePath;
        private readonly Mode mode;
        private readonly List<Employee> employees;
        private readonly EmployeeRepository repository;

        public FileTask(string filePath, Mode mode, List<Employee> employees, EmployeeRepository repository)
        {
            this.filePath = filePath;
            this.mode = mode;
            this.employees = employees;
            this.repository = repository;
        }

        public async Task RunAsync()
        {
            try
            {
                switch (mode)
                {
                    case Mode.READ:
                        var loaded = repository.LoadEmployees(filePath);
                        employees.Clear();
                        employees.AddRange(loaded);
                        break;

                    case Mode.WRITE:
                        repository.SaveEmployees(filePath, employees);
                        break;
                }
            }
            catch (AppException ex)
            {
                Console.Error.WriteLine($"FileTask Error ({mode}): {ex.Message}");
            }
        }
    }
}
